flannel.yaml

```
@@ -44,8 +44,62 @@ update:
     strip-prefix: v
 
 test:
+  environment:
+    contents:
+      packages:
+        - ${{package.name}}
+        - etcd
   pipeline:
-    # AUTOGENERATED
-    - runs: |
-        flanneld --version
-        flanneld --help
+    - name: "Test flanneld daemon output"
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          #!/bin/sh -ex
+          # Start etcd in background with proper networking configuration
+          mkdir -p /tmp/etcd-data
+          etcd --data-dir=/tmp/etcd-data \
+            --listen-client-urls=http://127.0.0.1:2379 \
+            --advertise-client-urls=http://127.0.0.1:2379 > /tmp/etcd.log 2>&1 &
+          ETCD_PID=$!
+
+          # Ensure etcd is ready before proceeding
+          for i in 1 2 3 4 5; do
+            if etcdctl --endpoints=http://127.0.0.1:2379 endpoint status --write-out=table; then
+              break
+            fi
+            sleep 1
+          done
+
+          # Store flannel network config in etcd
+          printf "{\"Network\": \"10.244.0.0/16\", \"Backend\": {\"Type\": \"host-gw\"}}" > /tmp/flannel-config.json
+          etcdctl --endpoints=http://127.0.0.1:2379 put /coreos.com/network/config "$(cat /tmp/flannel-config.json)"
+
+          # Save PID for cleanup
+          echo "export ETCD_PID=$ETCD_PID" > /tmp/etcd_pid.sh
+        # Disable iptables to avoid permission issues in unprivileged test environments and focus testing on core networking capabilities
+        start: flanneld --etcd-endpoints=http://127.0.0.1:2379 --iface=eth0 --iptables-forward-rules=false
+        timeout: 10
+        expected_output: |
+          Created subnet manager
+        error_strings: |
+          "msg":"error connecting to etcd"
+          "level":"error" .* "could not create subnet manager"
+          "panic"
+          "fatal"
+          "Failed"
+          "Error: could not access etcd"
+          "error: could not allocate lease"
+        post: |
+          #!/bin/sh -ex
+          # Load etcd PID from setup phase
+          . /tmp/etcd_pid.sh
+
+          # Verify flannel created the proper subnet configuration
+          if [ ! -f /run/flannel/subnet.env ]; then
+            echo "Subnet file not found"
+            exit 1
+          fi
+          grep -q "FLANNEL_SUBNET=" /run/flannel/subnet.env
+
```

```suggestion
+          # Check leases.
+          LEASE_SUBNET_ETCD_KEY=$(etcdctl get /coreos.com/network/subnets --prefix --keys-only)
+          LEASE_PUBLIC_IP=$(etcdctl get $LEASE_SUBNET_ETCD_KEY --print-value-only=true | jq '.PublicIP' | tr -d "\"")
+
+          LEASE_ETCD_LEASE_ID=$(etcdctl lease list | tail -1)
+          etcdctl lease timetolive --keys $LEASE_ETCD_LEASE_ID | grep $LEASE_SUBNET_ETCD_KEY
+          ip address show eth0 | grep $LEASE_PUBLIC_IP
+          ip route get $LEASE_PUBLIC_IP
```

This is an amazing test! Than ks @rahulduvedi!
I'd just add a small check on the IP leases it creates on etcd as etcd leases as well.